SIPS/sip-63.md

```diff
@@ -28,21 +28,80 @@ There is currently no way to track trading fees paid on-chain, so it is not poss
 
+**Trading Incentive Contract**
+
+When an exchange is made by a user on Synthetix Exchange, the amount of fees remitted (in sUSD) and user the funds are exchanged `from` will be recorded in the `Trading incentive contract`. Each `address` will accumulate the total of exchange fees that they generate for a period (tracked as a `periodID`) and the counter is reset when SNX for the incentives is deposited for the period and a user makes another trade. The total of the previous period will be stored on the next trade regardless of how long ago the last trade was.   
+
+For exchanges occuring through integrations and exchange aggregators such as `1inch` there will be a function that allows an `address` to be passed through the exchange to be recorded against the originating address for the trading volume incentives. 
+
+The `trading incentive` contract will track the `totalFeesInPeriod` for each period generated by exchanges, which will be used for calculating the % of the fees an account has generated in that period and hence the % of the SNX rewards they will be allocated. 
+
+The snx rewards an account can claim for a period is calculated as:
+
+\\[ reward = \frac{Account's Exchange Fees for period}{Total Exchange Fees for period} * SNX rewards for period 
+\\]
+
+The `totalFeesInPeriod` will be stored and kept for each period once the new SNX is deposited for that period and closes it, resetting the balances for the next period to accumulate. 
+
+This allows the account's to claim their trading incentives in any period they want in the future and combining multiple periods to claim.
+
+**Interface**
+
+```
+pragma solidity >=0.4.24;
+
+interface ITradingIncentive {
+  // Views
+  function rewards(uint periodID) external view; 
+  
+  function rewardForPeriods(uint[] periodIDs) external view;
+  
+  // Mutative Functions
+  function recordExchangeFee(uint amount, address account) external;
+  
+  function claimReward(uint periodID) external;
+  
+  function claimRewardForPeriods(uint[] periodIDs) external;
+  
+  // owner only
+  function notifyRewardAmount(uint reward) external;
+}
+```
+
+**Exchanger contract**
+
+Remit exchange fees generated by an account to the `Trading incentive` contract, storing the total fees for each account in the contract. 
+
+A function to allow trading incentives to be recorded against any `address` instead of the `msg.sender` of the `synthetix.exchange()` event. The `address` will need to be the one claiming the rewards from the `trading incentive` contract.    
+
 ### Test Cases
 <!--Test cases for an implementation are mandatory for SIPs but can be included with the implementation..-->
-Test cases for an implementation are mandatory for SIPs but can be included with the implementation.
+- When an exchange is made, the exchange fee in sUSD and the address the exchange `from` gets stored in the trading incentives contract and the `totalFeesInPeriod` is increased by the exchange fee amount. 
```

Comment:
It's worth being explicit and adding a test case when invoked via a third party/contract, etc.